### PR TITLE
Remove nd4j-native dep ensuring non cpu builds run

### PR DESCRIPTION
**Pull request by agibsonccc**
## What changes were proposed in this pull request?
1. Replaces nd4j-native dependency with nd4j-api for GPU builds,
2. fixes pi_build script defaults to enable jetson nano builds to be publishable.
3. Adds new version generator
(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

## Quick checklist

The following checklist helps ensure your PR is complete:

- [X ] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [X ] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [ X] Created tests for any significant new code additions.
- [X ] Relevant tests for your changes are passing.

### CODE REVIEW

File: contrib/version-updater/src/test/java/org/nd4j/AppTest.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+{
+    /**
+     * Rigorous Test :-)
+     */
+    @Test
+    public void shouldAnswerWithTrue()
+    {
+        assertTrue( true );
+    }
+}
```
Comment (treo) [inline contrib/version-updater/src/test/java/org/nd4j/AppTest.java]: I think we shouldn't commit the auto-generated non-tests.

File: libnd4j/pi_build.sh [hunk 2]
```diff
@@ -29,16 +29,16 @@ if [ -z "${HAS_ARMCOMPUTE}" ]; then export  HAS_ARMCOMPUTE=1; fi
 if [ -z "${ARMCOMPUTE_TAG}" ]; then export  ARMCOMPUTE_TAG=v20.05; fi
 if [ -z "${LIBND4J_BUILD_MODE}" ]; then export  LIBND4J_BUILD_MODE=Release; fi
 if [ -z "${ANDROID_VERSION}" ]; then export  ANDROID_VERSION=21; fi
-if [ -z "${CUDA_VER}" ]; then export  CUDA_VER=10.2; fi
+if [ -z "${v}" ]; then export  CUDA_VER=10.2; fi
```
Comment (treo) [inline libnd4j/pi_build.sh]: is this intentional?

**Review by treo — DISMISSED** [review]
The comments are mostly nitpicky.

File: libnd4j/pi_build.sh (hunk 2 above)
Comment (agibsonccc) [reply libnd4j/pi_build.sh]: No, fixed thanks. Great catch!

**Review by treo — APPROVED** [review]

**Pull request merged**